### PR TITLE
in qa:  bitcoin -> zcash

### PR DESCRIPTION
**Pull request by zancas**
Fixes: #4341

### CODE REVIEW

**Comment by zancas** [comment]
As a sanity check:

`./qa/zcash/full_test_suite.py` is run against a fresh build

All test suites _except_ rpc pass.

RPC is still running it might be helpful (efficient) to run it with a more powerful system than my laptop.

The aim of this PR is to replace all references to bitcoin *inside the qa* directory with references to zcash.

There are 2 places outside qa that had to be touched to facilitate this.

**Comment by zancas** [comment]
There remain a few references to filenames that include "bitcoin" _inside_ `qa` plus some correct comments that reference "bitcoin".

**Comment by zancas** [comment]
Sorry, the review request was premature.  Still debugging.

File: qa/pull-tester/run-zcashd-for-test.sh.in [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-# Copyright (c) 2013-2014 The Bitcoin Core developers
```
Comment (daira) [inline qa/pull-tester/run-zcashd-for-test.sh.in]: This is an incorrect change. It's fine to add copyright notices for "The Zcash Core developers", but notices for "The Bitcoin Core developers" should be retained in existing files or derived works of existing files.

File: qa/pull-tester/run-zcashd-for-test.sh.in [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
+# Copyright (c) 2013-2014 The Zcash Core developers
```
Comment (daira) [inline qa/pull-tester/run-zcashd-for-test.sh.in]: Update the Zcash copyright dates (but not the Bitcoin ones) to 2020.

**Comment by zancas** [comment]
I force pushed an update that I believe addresses all comments.

**Comment by zancas** [comment]
@mdr0id can you kick the CI pipeline to confirm that all the tests (especially the RPC tests!) pass against this feature branch?

**Comment by zancas** [comment]
Hey @oxarbitrage would you mind taking a look at this PR?
It's a refactor of the Python code that:

* replaces [B|b]itcoin with [Z|z]cash
* removal of whitespace in class names (style)

Only two files outside of `qa` are touched, they are build config files.

I'd like to get this landed because I've based a few other features on top of it.   In particular I'm hoping to migrate the RPC code to use `unittest.TestCase`, a change that I believe @daira is in favor of.

File: qa/rpc-tests/disablewallet.py [hunk 3]
```diff
@@ -23,7 +23,7 @@ def setup_network(self, split=False):
         self.sync_all()
 
     def run_test (self):
-        # Check regression: https://github.com/bitcoin/bitcoin/issues/6963#issuecomment-154548880
+        # Check regression: https://github.com/zcash/zcash/issues/6963#issuecomment-154548880
```
Comment (oxarbitrage) [inline qa/rpc-tests/disablewallet.py]: Link should not change.

File: qa/zcash/performance-measurements.sh [hunk 4]
```diff
@@ -357,9 +357,9 @@ case "$1" in
                 cat valgrind.out
                 rm -f valgrind.out
                 ;;
-            test_bitcoin)
+            test_zcash)
                 rm -f valgrind.out
```
Comment (oxarbitrage) [inline qa/zcash/performance-measurements.sh]: Here we have a problem. This pr is not changing the boost test binary(located at `zcash/src/test`) and with name `test_bitcoin`.

We could change the name of the binary however i am not totally sure how much is involved. For sure we need to change https://github.com/zcash/zcash/blob/master/src/test/Makefile and maybe https://github.com/zcash/zcash/blob/master/Makefile.am too.

File: qa/zcash/performance-measurements.sh [hunk 5]
```diff
@@ -357,9 +357,9 @@ case "$1" in
...
-                valgrind --leak-check=yes -v --error-limit=no --log-file="valgrind.out" ./src/test/test_bitcoin
+                valgrind --leak-check=yes -v --error-limit=no --log-file="valgrind.out" ./src/test/test_zcash
```
Comment (oxarbitrage) [inline qa/zcash/performance-measurements.sh]: Same as above, `test_zcash` will not be found.

**Comment by mdr0id** [comment]
@zancas Sure. I'd like to wait for the other builders to go into production first. Then run this against "all the things", so I don't have to manually test a bunch of stuff. We should have the new cross builders in CI production by EOD tomorrow

File: qa/zcash/performance-measurements.sh (hunk 4 above)
Comment (zancas) [reply qa/zcash/performance-measurements.sh]: Thanks for catching this, I tinkered with changing that name, but I think we should defer changes outside qa...

File: qa/rpc-tests/rawtransactions.py [hunk 6]
```diff
@@ -26,7 +27,7 @@ def setup_network(self, split=False):
         self.nodes = start_nodes(3, self.options.tmpdir)
 
         #connect to a local machine for debugging
-        #url = "http://bitcoinrpc:DP6DvqZtqXarpeNWyN3LZTFchCCyCUuHwNF7E8pX99x1@%s:%d" % ('127.0.0.1', 18232)
+        #url = "http://zcashrpc:DP6DvqZtqXarpeNWyN3LZTFchCCyCUuHwNF7E8pX99x1@%s:%d" % ('127.0.0.1', 18232)
         #proxy = AuthServiceProxy(url)
         #proxy.url = url # store URL on proxy for info
         #self.nodes.append(proxy)
```
Comment (daira) [inline qa/rpc-tests/rawtransactions.py]: Just delete this comment block.

File: qa/rpc-tests/rpcbind_test.py [hunk 7]
```diff
@@ -1,15 +1,16 @@
 #!/usr/bin/env python3
 # Copyright (c) 2014 The Bitcoin Core developers
+# Copyright (c) 2020 The Zcash developers
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or https://www.opensource.org/licenses/mit-license.php .
 
 # Test for -rpcbind, as well as -rpcallowip and -rpcconnect
 
-# Dependency: python-bitcoinrpc
+# Dependency: python-zcashrpc
```
Comment (daira) [inline qa/rpc-tests/rpcbind_test.py]: Is this a correct change?

File: qa/rpc-tests/test_framework/bignum.py [hunk 8]
```diff
@@ -2,7 +2,7 @@
 #
 # bignum.py
 #
-# This file is copied from python-bitcoinlib.
+# This file is copied from python-zcashlib.
```
Comment (daira) [inline qa/rpc-tests/test_framework/bignum.py]: Probably not correct.

File: qa/rpc-tests/test_framework/authproxy.py [hunk 9]
```diff
@@ -44,7 +44,7 @@
 
 HTTP_TIMEOUT = 600
 
-log = logging.getLogger("BitcoinRPC")
+log = logging.getLogger("ZcashRPC")
```
Comment (daira) [inline qa/rpc-tests/test_framework/authproxy.py]: Probably not correct.

File: qa/rpc-tests/test_framework/bignum.py [hunk 10]
```diff
@@ -81,7 +81,7 @@ def mpi2bn(s):
         return -v
     return v
 
-# bitcoin-specific little endian format, with implicit size
+# zcash-specific little endian format, with implicit size
```
Comment (daira) [inline qa/rpc-tests/test_framework/bignum.py]: Revert this change.

File: qa/rpc-tests/test_framework/test_framework.py [hunk 11]
```diff
@@ -5,7 +5,7 @@
 
 # Base class for RPC testing
 
-# Add python-bitcoinrpc to module search path:
+# Add python-zcashrpc to module search path:
```
Comment (daira) [inline qa/rpc-tests/test_framework/test_framework.py]: Is this correct?

File: qa/rpc-tests/test_framework/util.py [hunk 12]
```diff
@@ -7,7 +7,7 @@
 # Helpful routines for regression testing
 #
 
-# Add python-bitcoinrpc to module search path:
+# Add python-zcashrpc to module search path:
```
Comment (daira) [inline qa/rpc-tests/test_framework/util.py]: Is this correct?

File: qa/rpc-tests/turnstile.py [hunk 13]
```diff
@@ -26,22 +26,22 @@
...
-    bitcoind_processes
+    zcashd_processes
 )
 from decimal import Decimal
 
 TURNSTILE_ARGS = ['-experimentalfeatures',
                   '-developersetpoolsizezero']
 
-class TurnstileTest (BitcoinTestFramework):
+class TurnstileTest (ZcashTestFramework):
```
Comment (daira) [inline qa/rpc-tests/turnstile.py]:
```suggestion
class TurnstileTest(ZcashTestFramework):
```

Comment (daira) [reply qa/rpc-tests/turnstile.py]: And similarly for other classes below.

**Comment by daira** [comment]
Also I think the `ZCASHD` and `ZCASHCLI` env vars must have been set somewhere, otherwise it couldn't possibly have worked with "bitcoins" and "bitcoin-cli" as the defaults.

File: qa/pull-tester/rpc-tests.sh [hunk 14]
```diff
@@ -134,7 +134,7 @@ function runTestScript
     echo
 }
 
-if [ "x${ENABLE_BITCOIND}${ENABLE_UTILS}${ENABLE_WALLET}" = "x111" ]; then
+if [ "x${ENABLE_ZCASHD}${ENABLE_UTILS}${ENABLE_WALLET}" = "x111" ]; then
```
Comment (daira) [inline qa/pull-tester/rpc-tests.sh]: Is this change correct?

File: qa/rpc-tests/bip65-cltv-p2p.py [hunk 15]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env python3
 #
+# Copyright (c) 2020 The Zcash developers
```
[the remark this answers is not in the document]
Comment (zancas) [reply qa/rpc-tests/bip65-cltv-p2p.py]: Fixed.

File: qa/pull-tester/run-zcashd-for-test.sh [hunk 16]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (zancas) [reply qa/pull-tester/run-zcashd-for-test.sh]: Removed.

File: qa/rpc-tests/keypool.py [hunk 17]
```diff
@@ -1,15 +1,16 @@
 #!/usr/bin/env python3
 # Copyright (c) 2014 The Bitcoin Core developers
+# Copyright (c) 2020 The Zcash developers
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or https://www.opensource.org/licenses/mit-license.php .
 
 # Exercise the wallet keypool, and interaction with wallet encryption/locking
 
-# Add python-bitcoinrpc to module search path:
+# Add python-zcashrpc to module search path:
```
[the remark this answers is not in the document]
Comment (zancas) [reply qa/rpc-tests/keypool.py]: I don't understand the comment.   I'm not sure what it refers to.

Here's a chunk of `git annotate` output:

```
6c37f7fd78      (Wladimir J. van der Laan       2014-06-16 14:45:32 +0200       8)# Add python-bitcoinrpc to module search path:
aff0bf7fa1      (Daira Hopwood  2017-06-20 21:09:33 +0100       9)
aff0bf7fa1      (Daira Hopwood  2017-06-20 21:09:33 +0100       10)from test_framework.authproxy import JSONRPCException
aff0bf7fa1      (Daira Hopwood  2017-06-20 21:09:33 +0100       11)from test_framework.util import check_json_precision, initialize_chain,
\
aff0bf7fa1      (Daira Hopwood  2017-06-20 21:09:33 +0100       12)    start_nodes, start_node, stop_nodes, wait_bitcoinds, bitcoind_processes
aff0bf7fa1      (Daira Hopwood  2017-06-20 21:09:33 +0100       13)
6c37f7fd78      (Wladimir J. van der Laan       2014-06-16 14:45:32 +0200       14)import os
6c37f7fd78      (Wladimir J. van der Laan       2014-06-16 14:45:32 +0200       15)import sys
6c37f7fd78      (Wladimir J. van der Laan       2014-06-16 14:45:32 +0200       16)import shutil
6c37f7fd78      (Wladimir J. van der Laan       2014-06-16 14:45:32 +0200       17)import tempfile
6c37f7fd78      (Wladimir J. van der Laan       2014-06-16 14:45:32 +0200       18)import traceback
6c37f7fd78      (Wladimir J. van der Laan       2014-06-16 14:45:32 +0200       19)
```

It seems like @laanwj originally placed the comment before `import os`.   This doesn't make its meaning obvious to me.

[the remark this answers is not in the document]
Comment (zancas) [reply qa/rpc-tests/keypool.py]: I suppose the conservative thing to do is leave it untouched.

[the remark this answers is not in the document]
Comment (zancas) [reply qa/rpc-tests/keypool.py]: It seems that I have to rewrite history to restore it to its original state.   This is only a problem because reviewers will then need to re-review.

On the bright side I can also rebase atop the most recent master.

Since people have already done review-work I will not reset the comment to the original commit until I get approval from reviewers.

I believe that the options are:

(1)  revert the comment to the original text, but update the committer to me.  
(2)  rewrite the commit history such that the original text is comitted, and attributed to the original author
(3)  don't revert the change

Since I don't understand the comment, I think reverting to the original author is the best solution, but then there's a caveat about review work.

File: qa/rpc-tests/test_framework/bignum.py (hunk 10 above)
Comment (zancas) [reply qa/rpc-tests/test_framework/bignum.py]: Reverted

File: qa/rpc-tests/test_framework/test_framework.py (hunk 11 above)
Comment (zancas) [reply qa/rpc-tests/test_framework/test_framework.py]: Reverted with same caveats as above.  Should I rebase?

File: qa/rpc-tests/keypool.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (zancas) [reply qa/rpc-tests/keypool.py]: Digging a little deeper into the commit history:

```
git annotate master keypool.py | grep python-bitcoinrpc && git annotate master test_framework/test_framework.py | grep python-bitcoinrpc && git annotate master test_framework/util.py | grep python-bitcoinrpc
6c37f7fd78      (Wladimir J. van der Laan       2014-06-16 14:45:32 +0200       8)# Add python-bitcoinrpc to module search path:
e8097f7df1      (Gavin Andresen 2014-07-08 12:07:23 -0400       8)# Add python-bitcoinrpc to module search path:
356cfe8306      (Gavin Andresen 2014-02-26 16:31:18 -0500       10)# Add python-bitcoinrpc to module search path:
```

[the remark this answers is not in the document]
Comment (zancas) [reply qa/rpc-tests/keypool.py]: OK...   here's the line the comment almost certainly originally referred to:

https://github.com/bitcoin/bitcoin/blob/356cfe83065aa4f120707f026c10c83f1e62be89/qa/rpc-tests/util.py#L8

Then the comment got copied around, so I vote we remove it entirely (all three instances).

[the remark this answers is not in the document]
Comment (zancas) [reply qa/rpc-tests/keypool.py]: OK, this comment has been removed, which resolves this conversation.

File: qa/rpc-tests/test_framework/util.py (hunk 12 above)
Comment (zancas) [reply qa/rpc-tests/test_framework/util.py]: Resolved, see above.

File: qa/rpc-tests/test_framework/test_framework.py (hunk 11 above)
Comment (zancas) [reply qa/rpc-tests/test_framework/test_framework.py]: No.  It's obsolete, and removed.  See above.

File: qa/rpc-tests/test_framework/authproxy.py (hunk 9 above)
Comment (zancas) [reply qa/rpc-tests/test_framework/authproxy.py]: Reverted (to make logs consistent over time?).

File: qa/rpc-tests/test_framework/bignum.py (hunk 10 above)
Comment (zancas) [reply qa/rpc-tests/test_framework/bignum.py]: Ooops...   :( .

**Comment by zancas** [comment]
Because I can't bear to wait to find out if something is in conflict, I pushed this branch that contains the same code rebase on latest master:

https://github.com/zancas/zcash/tree/4341_rename_bitcoin_to_zcash_REBASED_ON_LATEST

File: qa/rpc-tests/keypool.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (daira) [reply qa/rpc-tests/keypool.py]: This is python-bitcoinrpc: https://github.com/jgarzik/python-bitcoinrpc

As far as I know, there is no python-zcashrpc. The dependency is still on python-bitcoinrpc.

File: qa/rpc-tests/rpcbind_test.py (hunk 7 above)
Comment (zancas) [reply qa/rpc-tests/rpcbind_test.py]: I suspect you're right, I will examine the git history.

File: qa/rpc-tests/test_framework/bignum.py (hunk 8 above)
Comment (zancas) [reply qa/rpc-tests/test_framework/bignum.py]: Will fix.

File: qa/rpc-tests/rpcbind_test.py (hunk 7 above)
Comment (zancas) [reply qa/rpc-tests/rpcbind_test.py]: Reverted.

File: qa/rpc-tests/keypool.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (zancas) [reply qa/rpc-tests/keypool.py]: The URL I posted above shows the code that updates the search path with `python-bitcoinrpc`, but that code is no-longer present, therefore the comment refers to code that's not there.   Therefore I deleted the comment.

Now that I think about it, I'm going to check whether `os` and `sys` are used elsewhere in that module.

[the remark this answers is not in the document]
Comment (zancas) [reply qa/rpc-tests/keypool.py]: Ok, they're both used elsewhere.

[the remark this answers is not in the document]
Comment (zancas) [reply qa/rpc-tests/keypool.py]: I believe this is resolved (by deleting the comment that references deleted code).

**Comment by zancas** [comment]
Hmmm...   I _think_ I've addressed all of @daira 's review suggestions (I also think she's super busy).

**Comment by zancas** [comment]
Let me know if there's anything I can do to expedite.

**Comment by zancas** [comment]
I don't know why this PR still shows "Changes requested".  I think all the requested changes have been applied.

@mdr0id any chance of running the CI pipeline against this?

**Comment by zancas** [comment]
Hi @daira I think this is ready to go in.   I'd like to improve the Python "on top" of this change.

**Comment by zancas** [comment]
This was left "out in the cold" for so long that it's stale.    Maybe I will redo the work someday.  For now I am closing the PR, @rex4539 might be interested to know that it makes his closed PR never-addressed.
